databases/jdp/jamo.go

```diff
@@ -76,26 +77,60 @@ type jamoPageQueryResponse struct {
 	Records     []jamoFileRecord `json:"records"`
 }
 
+// this flag is true until the first query to JAMO
+var jamoFirstQuery = true
+
+// this flag indicates whether JAMO is available (i.e. whether DTS is running
+// in the proper domain), and can only be trusted when jamoFirstQuery is false
+var jamoIsAvailable = false
+
+// this flag indicates whether we want to record JAMO queries (usually in a
+// testing environment, where it's set)
+var recordJamo = false
+
 // This function gathers and returns all jamo file records that correspond to
 // the given list of file IDs. The list of files is returned in the same order
 // as the list of file IDs.
 func queryJamo(fileIds []string) ([]jamoFileRecord, error) {
-	// fire up a recorder based on our testing needs (since we rely on JAMO
-	// for fetching file paths, and since JAMO only works within LBL's VPN)
-	var recordingMode recorder.Mode
-	if _, onVPN := os.LookupEnv("DTS_ON_LBL_VPN"); onVPN {
+	const jamoBaseUrl = "https://jamo-dev.jgi.doe.gov/"
+
+	if jamoFirstQuery {
+		// poke JAMO to see whether it's available in the current domain
+		resp, err := http.Get(jamoBaseUrl)
+		if err != nil {
+			return nil, err
+		}
+		if resp.StatusCode == http.StatusOK { // success!
+			jamoIsAvailable = true
+		}
+		jamoFirstQuery = false
+	}
+
+	// create a checksum that uniquely identifies the set of requested file IDs
+	checksum := md5.Sum([]byte(strings.Join(fileIds, ",")))
+
+	// set up a "VCR" to manage the recording and playback of JAMO queries
+	vcrMode := recorder.ModePassthrough // no recording or playback by default
```

Comment:
Since the VCR stuff is concerned with testing, can the VCR-related code just appear in the test files, rather than here in the main codebase?

I don't know how the go version of VCR is set up, but in the python version I create a 'context' for whatever functions or methods I'm testing within the test itself, e.g.

```py
    with vcr.use_cassette(
        "test/data/cassettes/fetch_objects_by_ref.yaml"
    ):
        output = data_fetcher.fetch_objects_by_ref(ref_list)
```
All the VCR settings are configured within the test suite and VCR is only active when the tests are running. The python VCR has request-matching capabilities so I can store the responses from a series of requests in a single file and the VCR module returns the appropriate response for the given request.

Reply:
Great idea. Do you think this should go into the current PR, or would it be okay for me to create an issue and fix it in another?

Reply:
Do it in another PR is fine with me.